src/compiler/utilities.ts

```diff
@@ -1667,6 +1667,27 @@ namespace ts {
         return SyntaxKind.FirstKeyword <= token && token <= SyntaxKind.LastKeyword;
     }
 
+    // Some keywords are TypeScript only, so they should not be parsed as considered as
+    // keywords in JavaScript
+    export function isJSKeyword(token: SyntaxKind): boolean {

```

Comment:
I was thinking maybe there will be TSX/JSX-only keywords in the future, then `isSolelyTypeScriptKeyword` wouldn't return the same thing as `isJSKeyword`. Though that might not likely to happen